src/services/TestResultsService.ts

```diff
@@ -597,6 +629,21 @@ export class TestResultsService {
   private static isValidTestCodeForExpiryCalculation(testCode: string): boolean {
     return TEST_CODES_FOR_CALCULATING_EXPIRY.CODES.includes(testCode);
   }
+
+  private static generatesExpiryOneYearMinusDay(payload: any, indexer: number): any {
+    payload.testTypes[indexer].testExpiryDate = dateFns.subDays(dateFns.addYears(new Date(), 1), 1).toISOString();
+    return payload;
+  }
+
+  private static generatesExpiryTwoYearsOnRegistrationDate(payload: any, indexer: number): any {
+    payload.testTypes[indexer].testExpiryDate = dateFns.addYears(payload.regnDate, 2).toISOString();
+    return payload;
+  }
+
+  private static checkEqualityOfExpiryAndRegDate(expiryDate: Date, regnDate: any): boolean {
+    const dateDefault = new Date(1970, 1, 1);
+    return dateFns.isEqual(expiryDate, dateDefault) && regnDate;
+  }
```

Comment:
What is this function doing? The name is a bit misleading

Reply:
Function name is confusing. The purpose is to check if most recent expiry is not found (it defaults to `new Date(1970,1,1)`